pallets/nfts/src/tests.rs

```diff
@@ -2188,6 +2885,207 @@ fn clear_all_transfer_approvals_works() {
 	});
 }
 
+#[test]
+fn clear_collection_approvals_works() {
+	new_test_ext().execute_with(|| {
+		let collection_id = 0;
+		let item_id = 42;
+		let owner = account(1);
+		let delegate_1 = account(3);
+		let delegate_2 = account(4);
+		let balance = 100;
+
+		Balances::make_free_balance_be(&owner, balance);
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			owner.clone(),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(account(1)),
+			collection_id,
+			item_id,
+			owner.clone(),
+			default_item_config()
+		));
+
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			delegate_1.clone(),
+			None
+		));
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			delegate_2.clone(),
+			None
+		));
+		// Remove zero collection approvals, no event emitted.
+		assert_ok!(Nfts::clear_collection_approvals(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			0
+		));
+		assert_eq!(Balances::free_balance(&owner), balance - 2);
+		assert_eq!(CollectionApprovals::<Test>::iter_prefix((0, owner.clone())).count(), 2);
+		assert!(!events().contains(&Event::<Test>::ApprovalsCancelled {
+			collection: collection_id,
+			item: None,
+			owner: owner.clone(),
+		}));
+
+		// Partially removes collection approvals.
+		assert_ok!(Nfts::clear_collection_approvals(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			1
+		));
+		assert_eq!(Balances::free_balance(&owner), balance - 1);
+		assert_eq!(CollectionApprovals::<Test>::iter_prefix((0, owner.clone())).count(), 1);
+
+		// Successfully remove all collection approvals.
+		assert_ok!(Nfts::clear_collection_approvals(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			2
+		));
+		assert!(events().contains(&Event::<Test>::ApprovalsCancelled {
+			collection: collection_id,
+			item: None,
+			owner: owner.clone(),
+		}));
+		assert_eq!(Balances::free_balance(&owner), balance);
+		assert!(CollectionApprovals::<Test>::iter_prefix((0, owner)).count().is_zero());
+
+		assert_noop!(
+			Nfts::transfer(RuntimeOrigin::signed(delegate_1), collection_id, item_id, account(5)),
+			Error::<Test>::NoPermission
+		);
+		assert_noop!(
+			Nfts::transfer(RuntimeOrigin::signed(delegate_2), collection_id, item_id, account(5)),
+			Error::<Test>::NoPermission
+		);
+	});
+}
+
+#[test]
+fn force_clear_collection_approvals_work() {
```

Comment:
This code can combine with the `clear_collection_approvals_works` test to reduce code duplication by having a for loop over the `[(Runtime::signed(origin), None), (Runtime::root(), Some(owner))]` but it need additional helper method. I'm not sure if it is better as keeping them in a separate test case keep it simple, but increase maintenance effort.

Reply:
Resolved in the next commit to follow this approach